src/components/DownloadTrajectoryMenu/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { Button, Tooltip } from "antd";
+import { ISimulariumFile } from "@aics/simularium-viewer/type-declarations";
+
+import { DATA_BUCKET_URL, TOOLTIP_COLOR } from "../../constants";
+import {
+    NetworkedSimFile,
+    LocalSimFile,
+    isNetworkSimFileInterface,
+} from "../../state/trajectory/types";
+import { Download } from "../Icons";
+
+import styles from "./style.css";
+
+interface DownloadTrajectoryMenuProps {
+    isBuffering: boolean;
+    simulariumFile: LocalSimFile | NetworkedSimFile;
+}
+
+const DownloadTrajectoryMenu = ({
+    isBuffering,
+    simulariumFile,
+}: DownloadTrajectoryMenuProps): JSX.Element => {
+    const [isDownloading, setIsDownloading] = React.useState(false);
+    const fileIsLoaded = () => !!simulariumFile.name;
+
+    const getHref = () => {
+        if (!fileIsLoaded()) {
+            return "";
+        }
+        if (isNetworkSimFileInterface(simulariumFile)) {
+            return `${DATA_BUCKET_URL}/trajectory/${simulariumFile.name}`;
+        } else {
+            const data: ISimulariumFile = simulariumFile.data;
+            // won't work if they loaded a binary file
+            const blob = data.getAsBlob();
+            return URL.createObjectURL(blob);
+        }
+    };
+
+    const downloadFile = async (fileName: string): Promise<void> => {
+        const downloadLink = document.createElement("a");
+        setIsDownloading(true);
+        downloadLink.download = fileName;
+        downloadLink.style.display = "none";
+        downloadLink.href = getHref();
+        document.body.appendChild(downloadLink);
+        downloadLink.click();
+        document.body.removeChild(downloadLink);
+        URL.revokeObjectURL(downloadLink.href);
+        setIsDownloading(false);
+    };
+
+    const onClick = () => {
+        if (!fileIsLoaded()) {
+            return;
+        }
+        downloadFile(simulariumFile.name);
```

Comment:
```suggestion
        setIsDownloading(true);
        await downloadFile(simulariumFile.name);
        setIsDownloading(false);
```

and remove them from the other function

Reply:
actually scratch this, I don't think it's ever possible to see the change in the tooltip, and that seems like the only reason for the `isDownloading` flag? When I click 'download' I never see it change it "Downloading..."

Reply:
@meganrm when you download a networked file do you get anything at all, or do you get a file that doesn't load properly in the viewer?

Reply:
I wasn't able to load a networked file. I just loaded a local file and then downloaded it again, which did work

Reply:
unfortunately this seems to be a bug with the viewer version 3.6.0. I revered it back (and removed the `getAsBlob`) and the networked files loaded. @toloudis or @ascibisz do you know why this is? Here are the changes in the new release: https://github.com/simularium/simularium-viewer/releases/tag/v3.6.0, and here is the error I'm getting 
``` 
[Debug] [netconnection] – "WS Connection Request Sent: " – "ws://staging-node1-agentviz-backend.cellexplore.net:9002/" (vendor.d1aa0b7eabb278739d2c.js, line 54046)

[Error] WebSocket connection to 'ws://staging-node1-agentviz-backend.cellexplore.net:9002/' failed: The network connection was lost.
```

Reply:
I suspect it's this one: https://github.com/simularium/simularium-viewer/pull/316 and maybe we need a patch release?

Reply:
yeah that's it. this issue wasn't showing up on my machine because i manually fixed the ws/wss issue in node modules yesterday after working with Alli.

Reply:
yeah, I tested it with main and it works. I also got this other error
```
Uncaught TypeError: Cannot read properties of undefined (reading 'sendWebSocketRequest')
```
So I added [a PR to fix it](https://github.com/simularium/simularium-viewer/pull/317), but I'm not sure why the websocketClient is undefined

Reply:
This has now all been merged in and I published a new version. You should update the viewer to 6.0.1 and then it should be good to go